DylibBinder/DBTypeDeclaration.cs

```diff
@@ -0,0 +1,200 @@
+			}
+			if (sb.Length > 0)
+				sb.Append ('>');
+			return sb.ToString ();
+		}
+
+		public List<DBAssociatedType> GetAssociatedTypes ()
+			=> Kind == TypeKind.Protocol ? GetAssociatedTypes (Funcs) : GetAssociatedTypes (Funcs, Properties);
+
+		static List<DBAssociatedType> GetAssociatedTypes (params IAssociatedTypes [] items)
+			=> items.SelectMany (t => t.AssociatedTypes.AssociatedTypeCollection).ToList ();
+	}
+
+	internal class DBTypeDeclarations {
+		public SortedDictionary<string, List<DBTypeDeclaration>> TypeDeclarationCollection { get; } = new SortedDictionary<string, List<DBTypeDeclaration>> ();
+		public static Dictionary<string, List<ClassContents>> InnerXDictionary { get; private set; } = new Dictionary<string, List<ClassContents>> ();
+
+		public DBTypeDeclarations (ModuleInventory mi) {
+			var checkInventory = new CheckInventoryDictionary (mi);
+			var innerX = new InnerX ();
+
+			foreach (var module in checkInventory.CheckInventoryDict.Keys) {
+				InnerXDictionary = innerX.AddClassContentsList (checkInventory.CheckInventoryDict[module].Classes,
+				                                                checkInventory.CheckInventoryDict [module].Enums,
+				                                                checkInventory.CheckInventoryDict [module].Structs);
+				FilterTypeDeclarations (module, checkInventory.CheckInventoryDict [module].Protocols,
+				                        checkInventory.CheckInventoryDict [module].Classes,
+				                        checkInventory.CheckInventoryDict [module].Structs,
+				                        checkInventory.CheckInventoryDict [module].Enums);
+			}
+		}
+
+		void FilterTypeDeclarations (string module, SortedSet<ProtocolContents> protocolContentList, params SortedSet<ClassContents>[] ClassContentListArray)
+		{
+			if (!TypeDeclarationCollection.ContainsKey (module))
+				TypeDeclarationCollection.Add (module, new List<DBTypeDeclaration> ());
+
+			var ignoredTypes = GetIgnoredTypes ();
+
+			foreach (var classContentsList in ClassContentListArray) {
+				foreach (var classContents in classContentsList) {
+					if (classContents.Name.ToFullyQualifiedName ().IsPublic () && InnerXDictionary.ContainsKey (classContents.Name.ToFullyQualifiedName ())
+						&& !ignoredTypes.Contains(classContents.Name.ToFullyQualifiedName ()))
+						TypeDeclarationCollection [module].Add (new DBTypeDeclaration (classContents));
+				}
+			}
+
+			foreach (var protocolContents in protocolContentList) {
+				if (protocolContents.Name.ToFullyQualifiedName ().IsPublic () && !ignoredTypes.Contains (protocolContents.Name.ToFullyQualifiedName ()))
+					TypeDeclarationCollection [module].Add (new DBTypeDeclaration (protocolContents));
+			}
+		}
+
+		SortedSet<string> GetIgnoredTypes ()
+		{
+			var fromTestPath = Path.Combine (Directory.GetCurrentDirectory (), "../../../../DylibBinder/IgnoreList.txt");
+			var fromVSPath = Path.Combine (Directory.GetCurrentDirectory (), "../../IgnoreList.txt");
```

Comment:
do they differ ?
if not then why not embed them into the assembly ?
you can still accept something else from the command-line

Reply:
> Do you mean to make this path an argument from the command the line?

No :)

You can include files inside an assembly (as resources [1]) and then get a `Stream` back to those at runtime. That way you do not have _find them_ in different locations at runtime. 

That's useful if you don't plan to modify the files without rebuilding them assembly (which the CI does every time). So unless it's meant to be user-edited (which I guess is why you also have a command-line option) then embedding your copy sounds useful.

[1] `csc` arguments to embed a file: https://github.com/xamarin/xamarin-macios/pull/11424/files

Reply:
The skip list(s) should be passed in as arguments in the command line.
The main target of this project is going to be `libswiftCore.dylib` (and any other swift-only library without a .swiftmodule file). I don't think it should ever be read in from static strings or an embedded resource.

Reply:
Okay I added the skip list to be an input argument!